iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java

```diff
@@ -203,4 +204,29 @@ public void testScanTableCaseInsensitive() throws IOException {
     Assert.assertArrayEquals(new Object[] {0L, "Alice", "Brown"}, rows.get(0));
     Assert.assertArrayEquals(new Object[] {1L, "Bob", "Green"}, rows.get(1));
   }
+
+  /**
+   * Column pruning could become problematic when a single Map Task contains multiple TableScan operators where
+   * different columns are pruned. This only occurs on MR, as Tez initializes a single Map task for every TableScan
+   * operator.
+   */
+  @Test
+  public void testMultiColumnPruning() throws IOException {
+    shell.setHiveSessionValue("hive.cbo.enable", true);
+
+    Schema schema1 = new Schema(optional(1, "fk", Types.StringType.get()));
+    List<Record> records1 = TestHelper.RecordsBuilder.newInstance(schema1).add("fk1").build();
+    testTables.createTable(shell, "table1", schema1, fileFormat, records1);
+
+    Schema schema2 = new Schema(optional(1, "fk", Types.StringType.get()), optional(2, "val", Types.StringType.get()));
+    List<Record> records2 = TestHelper.RecordsBuilder.newInstance(schema2).add("fk1", "val").build();
+    testTables.createTable(shell, "table2", schema2, fileFormat, records2);
+
+    // MR is needed for the reproduction
+    shell.setHiveSessionValue("hive.execution.engine", "mr");
```

Comment:
How's the test coverage for this on Tez? Is it already covered, or do we need to run this for Tez as well to avoid future regressions?

And a minor nit: we're setting the engine to MR but the test output will still show the parameter engine=Tez, which is not a big deal just hurts my OCD a bit :)

Reply:
With Hive 4.0.0 we do not support Iceberg. When I tried to run the tests with MR, the inserts were not working. So I had to run the inserts with Tez, and then the test query with MR.

OTOH this is a valid issue with MR, and older versions on Hive, where MR is supported. (Maybe on newer version as well)

Reply:
Got it. By the way, did you mean "With Hive 4.0.0 we do not support MR"?

Reply:
I am not sure that Hive removed MR support (I think not yet, but there is no active development maintaining it), but Hive - Iceberg definitely needs Tez.